src/AWS.Deploy.Orchestration/DisplayedResources/S3BucketResource.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using AWS.Deploy.Orchestration.Data;
+
+namespace AWS.Deploy.Orchestration.DisplayedResources
+{
+    public class S3BucketResource : IDisplayedResourceCommand
+    {
+        private readonly IAWSResourceQueryer _awsResourceQueryer;
+
+        public S3BucketResource(IAWSResourceQueryer awsResourceQueryer)
+        {
+            _awsResourceQueryer = awsResourceQueryer;
+        }
+
+        public async Task<Dictionary<string, string>> Execute(string resourceId)
+        {
+            string region = await _awsResourceQueryer.GetS3BucketLocation(resourceId);
+            string regionSeparator = ".";
+            if (string.Equals("us-east-1", region) ||
```

Comment:
This is where my previous comment about S3Region returning non standard aws region names for us-east-1 and eu-west-1 comes into play.

Reply:
Fixed